redbot/cogs/mutes/voicemutes.py

```diff
@@ -58,98 +66,161 @@ async def _voice_perm_check(
 
     @commands.command(name="voicemute")
     @commands.guild_only()
-    async def voice_mute(self, ctx: commands.Context, user: discord.Member, *, reason: str = None):
+    async def voice_mute(
+        self,
+        ctx: commands.Context,
+        users: commands.Greedy[discord.Member],
+        *,
+        time_and_reason: MuteTime = {},
+    ):
         """Mute a user in their current voice channel."""
-        user_voice_state = user.voice
-        if (
-            await self._voice_perm_check(
-                ctx, user_voice_state, mute_members=True, manage_channels=True
-            )
-            is False
-        ):
-            return
-        guild = ctx.guild
-        author = ctx.author
-        channel = user_voice_state.channel
-        audit_reason = get_audit_reason(author, reason)
-
-        success, issue = await self.mute_user(guild, channel, author, user, audit_reason)
-
-        if success["success"]:
-            await modlog.create_case(
-                self.bot,
-                guild,
-                ctx.message.created_at,
-                "vmute",
-                user,
-                author,
-                reason,
-                until=None,
-                channel=channel,
-            )
-            await ctx.send(
-                _("Muted {user} in channel {channel.name}").format(user=user, channel=channel)
-            )
-            try:
-                if channel.permissions_for(ctx.me).move_members:
-                    await user.move_to(channel)
+        if not users:
+            return await ctx.send_help()
+        if ctx.me in users:
+            return await ctx.send(_("You cannot mute me."))
+        if ctx.author in users:
+            return await ctx.send(_("You cannot mute yourself."))
+        async with ctx.typing():
+            success_list = []
+            issue_list = []
+            for user in users:
+                user_voice_state = user.voice
+
+                if (
+                    await self._voice_perm_check(
+                        ctx, user_voice_state, mute_members=True, manage_channels=True
+                    )
+                    is False
+                ):
+                    continue
```

Comment:
`_voice_perm_check()` sends an error message immediately which doesn't really play that well with how the rest of the command behave (as in, it returns errors at the end for the rest)